dali/operators/random/rng_base.h

```diff
@@ -53,18 +55,41 @@ class RNGBase : public Operator<Backend> {
       rng_ = rng;
     } else {
       static_assert(std::is_same_v<Backend, GPUBackend>);
-      DALI_FAIL("Checkpointing is not implemented for GPU random operators. ");
+      const auto &states_gpu = cpt.CheckpointState<curand_states>();
+      DALI_ENFORCE(states_gpu.length() == backend_data_.randomizer_.length(),
+                   "Provided checkpoint doesn't match the expected batch size. "
+                   "Perhaps the batch size setting changed? ");
+      backend_data_.randomizer_.set(states_gpu);
     }
   }
 
   std::string SerializeCheckpoint(const OpCheckpoint &cpt) const override {
-    const auto &state = cpt.CheckpointState<BatchRNG<std::mt19937_64>>();
-    return SnapshotSerializer().Serialize(state.ToVector());
+    if constexpr (std::is_same_v<Backend, CPUBackend>) {
+      const auto &state = cpt.CheckpointState<BatchRNG<std::mt19937_64>>();
+      return SnapshotSerializer().Serialize(state.ToVector());
+    } else {
+      static_assert(std::is_same_v<Backend, GPUBackend>);
+      const auto &states_gpu = cpt.CheckpointState<curand_states>();
+      size_t n = backend_data_.randomizer_.length();
```

Comment:
```suggestion
      size_t n = states_gpu.length();
```
Don't we want this^?